binr/rabin2/rabin2.c

```diff
@@ -112,25 +114,30 @@ static bool isBinopHelp(const char *op) {
 	return false;
 }
 
-static bool extract_binobj (const RBinFile *bf, const RBinObject *o, int idx) {
-	ut64 boffset = o ? o->boffset : 0;
-	ut64 bin_size = o ? o->obj_size : 0;
-	const ut8 *bytes;
+static bool extract_binobj (const RBinFile *bf, const RBinXtrData *data, int idx) {
+	ut64 boffset = data ? data->offset : 0;
+	ut64 bin_size = data ? data->size : 0;
+	char *bytes;
+	char *bytes_encoded;
 	//ut64 sz = bf ? r_buf_size (bf->buf) : 0;
-	RBinInfo *info = o ? o->info : NULL;
-	const char *arch = info ? info->arch : "unknown";
-	char bits = info ? info->bits : 0;
+	const char *arch = data->metadata ? data->metadata->arch : "unknown";
```

Comment:
check for data->metadata once, and set the defaults in the definition. conditionals are slow and better to avoid them if possible